File: Tutorial4-2Subcomponents/src/main/java/com/example/tutorial4_2subcomponents/ToastMaker.java
package com.example.tutorial4_2subcomponents;

import android.widget.Toast;

public class ToastMaker {

    private MainActivity mainActivity;

    private String message;

    public ToastMaker(MainActivity mainActivity) {
        this.mainActivity = mainActivity;
    }

    public void showToast(String message) {
        Toast.makeText(mainActivity, message, Toast.LENGTH_SHORT).show();
    }


    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }
}

File: Tutorial4-2Subcomponents/src/main/java/com/example/tutorial4_2subcomponents/model/ToastMaker.kt
package com.example.tutorial4_2subcomponents.model

import android.widget.Toast
import com.example.tutorial4_2subcomponents.MainActivity

class ToastMaker(private val mainActivity: MainActivity) {

    fun showToast(message: String?) {
        Toast.makeText(mainActivity, message, Toast.LENGTH_SHORT).show()
    }

}